icu4c/source/test/intltest/ucdtest.cpp

```diff
@@ -1024,3 +1026,174 @@ void UnicodeTest::TestIDCompatMath() {
+    assertEquals("ID_Type(MUSICAL SYMBOL KIEVAN C CLEF)", u"Not_XID Technical Uncommon_Use", getIDTypes(0x1D1DE));
+    assertEquals("ID_Type(MRO LETTER TA)", u"Exclusion Uncommon_Use", getIDTypes(0x16A40));
+    assertEquals("ID_Type(GREEK MUSICAL LEIMMA)", u"Not_XID Obsolete", getIDTypes(0x1D245));
+
+    // error handling
+    UIdentifierType types[2];
+    UErrorCode failure = U_ZERO_ERROR;
+    u_getIDTypes(0, types, -1, &failure);
+    assertEquals("u_getIDTypes(capacity<0)", U_ILLEGAL_ARGUMENT_ERROR, failure);
+
+    failure = U_ZERO_ERROR;
+    u_getIDTypes(0, nullptr, 1, &failure);
+    assertEquals("u_getIDTypes(nullptr)", U_ILLEGAL_ARGUMENT_ERROR, failure);
+
+    failure = U_ZERO_ERROR;
+    int32_t length = u_getIDTypes(0x30, types, 0, &failure);
+    assertEquals("u_getIDTypes(digit 0, capacity 0) overflow", U_BUFFER_OVERFLOW_ERROR, failure);
+    assertEquals("u_getIDTypes(digit 0, capacity 0) length", 1, length);
+
+    failure = U_ZERO_ERROR;
+    length = u_getIDTypes(0x1D1DE, types, 0, &failure);
+    assertEquals("u_getIDTypes(Kievan C clef, capacity 2) overflow", U_BUFFER_OVERFLOW_ERROR, failure);
+    assertEquals("u_getIDTypes(Kievan C clef, capacity 2) length", 3, length);
+
+    // Property names work and get the correct sets.
+    UnicodeSet rec(u"[:Identifier_Type=Recommended:]", errorCode);
+    UnicodeSet incl(u"[:Identifier_Type=Inclusion:]", errorCode);
+    UnicodeSet limited(u"[:Identifier_Type=Limited_Use:]", errorCode);
+    UnicodeSet uncommon(u"[:Identifier_Type=Uncommon_Use:]", errorCode);
+    UnicodeSet notChar(u"[:Identifier_Type=Not_Character:]", errorCode);
+    // Unicode 15.1 set sizes; normally grows over time except Not_Character shrinks
+    assertTrue("Recommended number of characters", rec.size() >= 112761);
+    assertTrue("Inclusion number of characters", incl.size() >= 17);
+    assertTrue("Limited_Use number of characters", limited.size() >= 5268);
+    assertTrue("Uncommon_Use number of characters", uncommon.size() >= 398);
+    assertTrue("Not_Character number of characters",
+               800000 <= notChar.size() && notChar.size() <= 964293);
+    assertFalse("Recommended.constains(slash)", rec.contains(0x2F));
+    assertTrue("Recommended.constains(digit 0)", rec.contains(0x30));
+    assertTrue("Inclusion.constains(colon)", incl.contains(0x3A));
+    assertTrue("Recommended.constains(U+2EE5D)", rec.contains(0x2EE5D));
+    assertTrue("Limited_Use.constains(SYRIAC FEMININE DOT)", limited.contains(0x0740));
+    assertTrue("Limited_Use.constains(NKO LETTER JONA JA)", limited.contains(0x7E8));
+    assertTrue("Not_Character.constains(surrogate)", notChar.contains(0xd800));
+    assertTrue("Not_Character.constains(U+10FFFF)", notChar.contains(0x10FFFF));
+    assertTrue("Uncommon_Use.constains(LATIN SMALL LETTER EZH)", uncommon.contains(0x0292));
+    assertTrue("Uncommon_Use.constains(MUSICAL SYMBOL KIEVAN C CLEF)", uncommon.contains(0x1D1DE));
```

Comment:
s/constains/contains throughout this section

Reply:
oh my =-)